### PR TITLE
netty 4.1 upgrade

### PR DESCRIPTION
**Pull request by yschimke**
Upgrade to Netty 4.1 with the minimal set of changes

### CODE REVIEW

File: src/main/java/com/twitter/http2/DefaultHttp2Headers.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import io.netty.handler.codec.http.DefaultHttpHeaders;
+import io.netty.util.AsciiString;
+import io.netty.util.ByteProcessor;
+import io.netty.util.internal.PlatformDependent;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+public class DefaultHttp2Headers extends DefaultHttpHeaders {
+  private static final ByteProcessor HEADER_NAME_VALIDATOR = new ByteProcessor() {
+    @Override
```
Comment (madster26) [inline src/main/java/com/twitter/http2/DefaultHttp2Headers.java:13]: Doesn't all of this come for free when we use super(true)?

Comment (yschimke) [inline src/main/java/com/twitter/http2/DefaultHttp2Headers.java:13]: No it rejects ':' in the key names

Comment (madster26) [inline src/main/java/com/twitter/http2/DefaultHttp2Headers.java:13]: Ah ok I missed the ":" so this is different then from the DefaultHttp2Headers in the main netty project where they don't check all of this? Do we know why?

File: src/main/java/com/twitter/http2/DefaultHttp2Headers.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      case '\n':
+      case 0x0b:
+      case '\f':
+      case '\r':
+      case ' ':
+      case ',':
+      case ';':
+      case '=':
+        throw new IllegalArgumentException(
+                "a header name cannot contain the following prohibited characters: =,; \\t\\r\\n\\v\\f: " +
```
Comment (madster26) [inline src/main/java/com/twitter/http2/DefaultHttp2Headers.java:32]: Perhaps drop the : in the Exception message?

Comment (yschimke) [inline src/main/java/com/twitter/http2/DefaultHttp2Headers.java:32]: Good catch, the exception in netty 4.1 lists it twice

```
codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java 
"a header name cannot contain the following prohibited characters: =,;: \\t\\r\\n\\v\\f: " +
```

Comment (yschimke) [inline src/main/java/com/twitter/http2/DefaultHttp2Headers.java:32]: Actually, it is a : separating the explanation from the bad character.  Not sure if I should remove e.g. just below

```
    if (character > 127) {
      throw new IllegalArgumentException("a header name cannot contain non-ASCII character: " +
              character);
    }
```

**Comment by Scottmitch** [comment]
Hey guys have you seen https://github.com/twitter/netty-http2/issues/3? It would be great if you are finding issues if we could work together and push the fixes upstream.

**Pull request merged**